proto/regen/ecocredit/v1/state.proto

```diff
@@ -236,3 +244,24 @@ message BatchSupply {
   // amount_cancelled will always equal the original credit issuance amount.
   string cancelled_amount = 4;
 }
+
+// BatchOrigTx will index batch mint operations using orig tx ID to handle
+// potential double minting errors
+message BatchOrigTx {
+  option (cosmos.orm.v1alpha1.table) = {
+    id : 11,
+    primary_key : {fields : "tx_id"}
+  };
+
+  // the id of an originating transaction or opeartion.
+  string tx_id = 1;
+
+  // type of the transaction originating the mint process. Eg: Polygon,
+  // Ethereum, Verra,
+  string typ = 2;
+
+  string note = 3;
+
+  // batch_denom is the unique ID of the credit batch.
+  string batch_denom = 4;
```

Comment:
Normally I would prefer to not storing everything. We have events to keep that information. But events are not guaranteed to be permanent.
So ideally we don't store everything here and we sort out the events in SDK.

What do you think?

Reply:
in fact, since I'm adding this fields here, I've slimmed down the `EventMintBatchCredits` event